### ISSUE TITLE
docs(reference): fix config-core.md entry point and add a Bring your own Hydra config section

### ISSUE THREAD

**Issue by CedrickArmel** [CedrickArmel/radiologist#242]
Part of the docs accuracy epic. **Content only** — do not change any Python, config, or CLI code in this issue.

Target file: `docs/reference/config-core.md` (97 lines, hand-authored — **not** an include shim, so it must be edited directly)

---

### 1. The stated entry point does not exist (lines 4–16)

> The Hydra entry point is `radiologist.core.train:main`
> (`python -m radiologist.core.train`); it loads `configs/train.yaml` by
> default, or another top-level config via `--config-name` (e.g. `eval`).

The example that follows invokes `uv run --active python -m radiologist.core.train` with `trainer.max_epochs=30 module/optimizer=adamw datamodule.batch_size=16`.

`radiologist-core/src/radiologist/core/train.py` has no `main`, no `@hydra.main`, and no `__main__` block. The real entry point is `radiologist.cli.groups.core:train_main` at `radiologist-cli/src/radiologist/cli/groups/core.py:101`, reached as `radiologist core`.

### 2. The `onnx_export.yaml` row is wrong (`## callbacks` table)

It claims the callback is "opt-in, not part of `default.yaml`'s composition; must be added explicitly". The actual defaults list in `radiologist-core/src/radiologist/core/configs/callbacks/default.yaml` composes six entries: `best_metric`, `wandb_summary`, `attribution`, `model_checkpoint`, `lr_monitor`, `onnx_export`.

It **is** composed by default. The `default.yaml` row in the same table also lists only five callbacks and needs `onnx_export` appended.

### 3. A dangling cross-reference points here

`docs/reference/environment.md:77` links to "the 'bring your own Hydra config' section of [`config-core.md`](config-core.md)". No such section exists — this file's nine headings are `## Top-level singletons`, `## callbacks/`, `## module/`, `## datamodule/`, `## debug/`, `## loggers/`, `## strategy/`, `## hydra/`. `mkdocs.yml`'s `validation.links.not_found: info` does not catch this because it is an anchor-less link to a file that does exist. AC2 below repairs it by creating the section under exactly that name — no edit to `environment.md` is needed.

---

### AC1 — Replace lines 4–16 (opening paragraph + example)

````markdown
`radiologist-core` composes its training/evaluation run entirely from a
[Hydra](https://hydra.cc) config tree at `radiologist-core/src/radiologist/core/configs/`.
The Hydra entry point is `radiologist.cli.groups.core:train_main`, shipped in the
`radiologist-cli` package and reached as the `core` group of the unified
`radiologist` command. It loads `configs/train.yaml` by default, or another
top-level config via `--config-name` (e.g. `eval`).

Everything after `core` is forwarded to Hydra verbatim, so every group below can
be overridden on the command line:

```bash
radiologist core \
    trainer.max_epochs=30 \
    module/optimizer=adamw \
    datamodule.batch_size=16
```

```bash
radiologist core --config-name eval ckpt_path=/path/to/checkpoint.ckpt
```

The `core` group backs onto a hard dependency, so plain `pip install radiologist-cli`
is enough — unlike `etl`, `registry`, and `infer`, it needs no extra.
````

### AC2 — New `## Bring your own Hydra config`

Append at the **end of the file**, after `## hydra/`. The heading text must be exactly `## Bring your own Hydra config` so that the `environment.md:77` reference resolves.

`````markdown
## Bring your own Hydra config

The config tree above ships **inside the installed wheel** — `train_main` is
declared with `config_path="pkg://radiologist.core.configs"`, not a filesystem
path. You extend it by giving Hydra an additional search path rather than by
editing the installed package.

```bash
# as a config override
radiologist core hydra.searchpath=[file:///abs/path/to/myconfigs] module/loss=my_loss

# as the equivalent Hydra CLI flag
radiologist core --config-dir /abs/path/to/myconfigs module/loss=my_loss
```

Mirror the packaged group layout inside your directory:

```
myconfigs/
├── module/
│   ├── my_net.yaml            # -> module=my_net
│   ├── loss/my_loss.yaml      # -> module/loss=my_loss
│   ├── metric/my_metric.yaml  # -> module/metric=my_metric
│   ├── optimizer/my_opt.yaml  # -> module/optimizer=my_opt
│   └── scheduler/my_sched.yaml
├── callbacks/my_callbacks.yaml
└── datamodule/my_datamodule.yaml
```

Each file needs the same `# @package` header its packaged siblings use:

| Group | Required header |
|---|---|
| `module`, `module/loss`, `module/metric`, `module/optimizer`, `module/scheduler` | `# @package module` |
| `callbacks/*` | `# @package callbacks` |
| `datamodule/*`, `trainer.yaml`, `debug/*` | `# @package _global_` |

Omitting the header is the most common failure: Hydra infers the package from
the directory name, your `loss:` key lands at `module.loss.loss`, and `LModule`
— which reads `cfg.loss` — never sees it. The symptom is an instantiation error
deep in the run, not a config error at compose time. Check with
`radiologist core --cfg job` before launching anything expensive; it prints the
fully composed config and exits.

A minimal custom loss:

```yaml
# myconfigs/module/loss/my_loss.yaml
# @package module
loss:
  _target_: my_package.losses.LabelSmoothedCE
  smoothing: 0.05
```

`_target_` may point at any importable class — your package does not have to be
part of this workspace, it only has to be on `sys.path` of the interpreter
running `radiologist`.

See [radiologist-core/README.md](../pkg/core.md) for the per-key contract each
group member must satisfy (which entries need `_partial_: true`, what `LModule`
calls each collaborator with, and the `${module.net.num_classes}` interpolation
you must keep intact when swapping the backbone).
`````

### AC3 — Fix the two `## callbacks/` table rows

- `default.yaml` row: append `onnx_export` to the list of composed callbacks (six, not five).
- `onnx_export.yaml` row: replace the opening clause with —

  > `radiologist.core.OnnxExportCallback` — part of `default.yaml`'s composition. It is a silent no-op without an active W&B run and a best checkpoint, which is why it is safe to leave in the default list. `input_shape: [1, 3, 224, 224]`, `classes: ["normal", "abnormal"]` (placeholder values — override to match the actual class list), `cam_target_layer: layer4`, `opset: 18`. Exports ONNX artifacts at fit end.

---

## Do NOT change

I spot-verified the `## module/`, `## datamodule/`, `## debug/`, `## loggers/`, `## strategy/`, `## hydra/` and `## Top-level singletons` tables against the yaml on disk — they are accurate. In particular, **keep** the honest "flagged as unclear" note in the `debug/limit.yaml` and `debug/overfit.yaml` rows about `early_stopping`/`exception_checkpoint` overrides referencing callbacks that do not exist; that observation is correct and useful.

## Verification

The `hydra.searchpath` and `--config-dir` mechanisms in AC2 were **executed** against the real CLI: a custom `module/loss/my_loss.yaml` in an external directory composed correctly via both forms, and omitting the `# @package module` header reproduced exactly the silent misplacement the section warns about. `radiologist core --config-name eval` was also executed and produced `stage: eval`, `train: false`.

## Definition of done

- [ ] All three ACs applied to `docs/reference/config-core.md`
- [ ] The `## Bring your own Hydra config` heading text is exact (repairs the `environment.md:77` link)
- [ ] No source, config, or CLI file changed; `environment.md` untouched
- [ ] `mkdocs build` succeeds (`make docs-build`)
